### PR TITLE
Queries and update animals table

### PR DESCRIPTION
**Pull request by arnoldnekemiah**
-Made more Queries and Altered the` animals` table as suggested.
-Made use of Transactions to play with the animal table.
-Made use of Aggregate functions.

Below are the Screenshots for the results;

update1

update2

update3

update4

update5

image

update7

update8

update9

update10

update11

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -9,3 +9,22 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
```
Comment (Deepakdanger) [inline queries.sql:18]: -[ ] Here need 2 more transaction
1. - UPDATE the animals table by setting the species column to digimon for all animals that have a name ending in mon.

  - UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Commit the transaction.
```
   BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```
2. Inside a transaction delete all records in the animals table, then roll back the transaction.

```sql
   BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

File: queries.sql [hunk 2]
```diff
@@ -9,3 +9,22 @@ SELECT * FROM animals WHERE neutered = true;
...
+SAVEPOINT my_savepoint1;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO my_savepoint1;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts FROM animals GROUP BY neutered;
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (Deepakdanger) [inline queries.sql:38]: - Here we have to run more query .
- [ ] What is the average weight of animals?
- [ ] What is the minimum and maximum weight of each type of animal?
- kindly attach a screenshot of this also

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @arnoldnekemiah , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql
Comment (AbelG101) [inline queries.sql:1]: - [ ] As mentioned previously please make sure to update the screenshots on the PR description with the new query results.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @arnoldnekemiah 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correctly made changes to queries ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great summary ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql
Comment (arnoldnekemiah) [reply queries.sql:1]: Hey @AbelG101, I updated the screenshots check the last two screenshots

Comment (arnoldnekemiah) [reply queries.sql:1]: I had forgotten to click update

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @arnoldnekemiah ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Added screenshots to the pull request description ✔️
- Created queries to answer the questions in
transactions ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**